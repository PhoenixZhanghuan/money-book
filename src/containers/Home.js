import React, { Component, Fragment } from "react";
import logo from "../logo.svg";
import "bootstrap/dist/css/bootstrap.min.css";
import "../App.css";
import PriceList from "../components/PriceList";
import { LIST_VIEW, CHART_VIEW, TYPE_INCOME, TYPE_OUTCOME } from "../utility";
import ViewTab from "../components/ViewTab";
import MonthPicker from "../components/MonthPicker";
import TotalPrice from "../components/TotalPrice";
import CreateBtn from '../components/CreateBtn';

const items = [
    {
        id: 1,
        title: "去云南旅行",
        price: 200,
        date: "2018-09-10",
        category: {
            id: 1,
            name: "旅行",
            type: "outcome",
            iconName: "ios-plane",
        },
    },
    {
        id: 2,
        title: "去云南旅行",
        price: 400,
        date: "2018-10-10",
        category: {
            id: 1,
            name: "旅行",
            type: "outcome",
            iconName: "ios-plane",
        },
    },
    {
        id: 3,
        title: "理财收入",
        price: 200,
        date: "2018-11-10",
        category: {
            id: 1,
            name: "理财",
            type: "income",
            iconName: "logo-yen",
        },

    }
];

class Home extends Component {
    render() {
        let totalIncome = 0,
            totalOutcome = 0;
        items.forEach((item) => {
            if (item.category.type === TYPE_OUTCOME) {
                totalOutcome += item.price;
            } else {
                totalIncome += item.price;
            }
        });
        return (
            <Fragment>
                <header className="App-header">
                    <div className="row mb-5">
                        <img src={logo} className="App-logo" alt="logo" />
                    </div>
                    <div className="row">
                        <div className="col">
                            <MonthPicker
                                year={2018}
                                month={8}
                                onChange={() => {}}
                            />
                        </div>
                        <div className="col">
                            <TotalPrice
                                income={totalIncome}
                                outcome={totalOutcome}
                            />
                        </div>
                    </div>
                </header>
                <div className="content-area py-3 px-3">
                    <ViewTab activeTab={LIST_VIEW} onTabChange={() => {}}/>
                    <CreateBtn onClick={() => {}} />
                    <PriceList 
                        items={items}
                        onModifyItem={() => {}}
                        onDeleteItem={() => {}}
                    />
                </div>
            </Fragment>
        );
    }
}

export default Home;
