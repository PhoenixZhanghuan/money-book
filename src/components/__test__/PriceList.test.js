import React from 'react';
import { shallow } from 'enzyme';
import Ionicon from 'react-ionicons';
import PriceList from '../PriceList';
import {items, categories} from '../../containers/Home';

const itemsWithCategory = items.map(item => {
    item.category = categories[item.cid]
    return item;
})

const props = {
    items: itemsWithCategory,
    onModifyItem: jest.fn(),
    onDeleteItem: jest.fn()
}

let wrapper; 
describe('test PriceList component', () => {
    beforeEach(() => {
        wrapper = shallow(<PriceList {...props} />)
    })

    it('should reander the component to match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })

    it('should render correct price items length', () => {
        expect(wrapper.find('.list-group-item').length).toEqual(itemsWithCategory.length)
    })

    it('should render corrent icon and price for each item', () => {
        const iconList = wrapper.find('.list-group-item').first().find(Ionicon)
        expect(iconList.length).toEqual(3);
        expect(iconList.first().props().icon).toEqual(itemsWithCategory[0].category.iconName)
    })
     
    it('should trigger the correct function callbacks', () => {
        const firstItem = wrapper.find('.list-group-item').first();
        firstItem.find('a').first().simulate('click')
        expect(props.onModifyItem).toHaveBeenCalledWith(itemsWithCategory[0]);
        firstItem.find('a').last().simulate('click');
        expect(props.onDeleteItem).toHaveBeenCalledWith(itemsWithCategory[0]);
    })
})

