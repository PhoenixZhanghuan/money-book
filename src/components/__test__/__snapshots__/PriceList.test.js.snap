// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test PriceList component should reander the component to match snapshot 1`] = `
<ul
  className="list-groud list-group-flush"
>
  <li
    className="list-group-item d-flex justify-content-between align-items-center"
    key="1"
  >
    <span
      className="col-1 badge badge-primary"
    >
      <Ionicon
        beat={false}
        className="rounded-circle"
        color="#fff"
        fontSize="30px"
        icon="ios-plane"
        rotate={false}
        shake={false}
        style={
          Object {
            "backgroundColor": "#007bff",
            "padding": "5px",
          }
        }
      />
    </span>
    <span
      className="col-5"
    >
      去云南旅行
    </span>
    <span
      className="col-2 font-weight-bold"
    >
      -
      200
      元
    </span>
    <span
      className="col-2"
    >
      2021-04-10
    </span>
    <a
      className="col-1"
      onClick={[Function]}
    >
      <Ionicon
        beat={false}
        className="rounded-circle"
        color="#fff"
        fontSize="30px"
        icon="ios-create-outline"
        rotate={false}
        shake={false}
        style={
          Object {
            "backgroundColor": "#28a745",
            "padding": "5px",
          }
        }
      />
    </a>
    <a
      className="col-1"
      onClick={[Function]}
    >
      <Ionicon
        beat={false}
        className="rounded-circle"
        color="#fff"
        fontSize="30px"
        icon="ios-close"
        rotate={false}
        shake={false}
        style={
          Object {
            "backgroundColor": "#dc3545",
            "padding": "5px",
          }
        }
      />
    </a>
  </li>
  <li
    className="list-group-item d-flex justify-content-between align-items-center"
    key="2"
  >
    <span
      className="col-1 badge badge-primary"
    >
      <Ionicon
        beat={false}
        className="rounded-circle"
        color="#fff"
        fontSize="30px"
        icon="ios-plane"
        rotate={false}
        shake={false}
        style={
          Object {
            "backgroundColor": "#007bff",
            "padding": "5px",
          }
        }
      />
    </span>
    <span
      className="col-5"
    >
      去云南旅行
    </span>
    <span
      className="col-2 font-weight-bold"
    >
      -
      400
      元
    </span>
    <span
      className="col-2"
    >
      2021-03-10
    </span>
    <a
      className="col-1"
      onClick={[Function]}
    >
      <Ionicon
        beat={false}
        className="rounded-circle"
        color="#fff"
        fontSize="30px"
        icon="ios-create-outline"
        rotate={false}
        shake={false}
        style={
          Object {
            "backgroundColor": "#28a745",
            "padding": "5px",
          }
        }
      />
    </a>
    <a
      className="col-1"
      onClick={[Function]}
    >
      <Ionicon
        beat={false}
        className="rounded-circle"
        color="#fff"
        fontSize="30px"
        icon="ios-close"
        rotate={false}
        shake={false}
        style={
          Object {
            "backgroundColor": "#dc3545",
            "padding": "5px",
          }
        }
      />
    </a>
  </li>
  <li
    className="list-group-item d-flex justify-content-between align-items-center"
    key="3"
  >
    <span
      className="col-1 badge badge-primary"
    >
      <Ionicon
        beat={false}
        className="rounded-circle"
        color="#fff"
        fontSize="30px"
        icon="logo-yen"
        rotate={false}
        shake={false}
        style={
          Object {
            "backgroundColor": "#007bff",
            "padding": "5px",
          }
        }
      />
    </span>
    <span
      className="col-5"
    >
      理财收入
    </span>
    <span
      className="col-2 font-weight-bold"
    >
      +
      200
      元
    </span>
    <span
      className="col-2"
    >
      2021-02-10
    </span>
    <a
      className="col-1"
      onClick={[Function]}
    >
      <Ionicon
        beat={false}
        className="rounded-circle"
        color="#fff"
        fontSize="30px"
        icon="ios-create-outline"
        rotate={false}
        shake={false}
        style={
          Object {
            "backgroundColor": "#28a745",
            "padding": "5px",
          }
        }
      />
    </a>
    <a
      className="col-1"
      onClick={[Function]}
    >
      <Ionicon
        beat={false}
        className="rounded-circle"
        color="#fff"
        fontSize="30px"
        icon="ios-close"
        rotate={false}
        shake={false}
        style={
          Object {
            "backgroundColor": "#dc3545",
            "padding": "5px",
          }
        }
      />
    </a>
  </li>
</ul>
`;
